### PR TITLE
Updater enable updates to 9.1.3, 9.0.7

### PR DESCRIPTION
**Pull request by PVince81**
#### Minor updates
- [x] TEST: 9.0.6 -> 9.0.7
- [x] TEST: 9.1.2 -> 9.1.3

#### Major updates
- [ ] TEST: 8.2.x -> 9.0.7
- [ ] TEST: 9.0.7 -> 9.1.3

@davitol @VicDeo

Please note that we moved the updater to this public repo so don't be confused where to find this PR.

### CODE REVIEW

**Comment by PVince81** [comment]
Adjusted Makefile and removed deps.

When installing the server one now needs to run `make` (but there are no composer deps currently) and to run tests `make test` will run both unit tests and integration tests.

**Comment by davitol** [comment]
Updating from 9.0.7 to 9.1.3. via daily channel

The following integrity check warnings appear:
```
Results
=======
- core
	- INVALID_HASH
		- updater/app/config/container.php
		- updater/src/Command/CheckSystemCommand.php
		- updater/src/Command/CheckpointCommand.php
		- updater/src/Command/ExecuteCoreUpgradeScriptsCommand.php
		- updater/src/Console/Application.php
		- updater/src/Controller/IndexController.php
		- updater/src/Utils/AppManager.php
		- updater/src/Utils/Checkpoint.php
		- updater/src/Utils/Fetcher.php
		- updater/src/Utils/Locator.php
		- updater/src/Utils/ZipExtractor.php
	- FILE_MISSING
		- updater/src/Resources/views/partials/error.php
		- updater/src/Tests/Utils/DocLinkTest.php
		- updater/src/Utils/DocLink.php
	- EXTRA_FILE
		- updater/src/Utils/BzipExtractor.php
		- updater/src/Utils/ConfigWriter.php
```

@PVince81 @VicDeo

**Comment by PVince81** [comment]
😢 @VicDeo

**Comment by davitol** [comment]
Updating from 8.2.8 to 9.0.7 via daily channel

The following integrity check warnings appear:

```
Results
=======
- provisioning_api
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.

Raw output
==========
Array
(
    [provisioning_api] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

)
```

**Comment by PVince81** [comment]
@VicDeo please have a look at these issues

File: update-server/tests/integration/features/update.beta.feature [hunk 1]
```diff
@@ -2,33 +2,33 @@ Feature: Testing the update scenario of releases on the beta channel
...
+  Scenario: Updating an outdated ownCloud 9.0.7 on the beta channel
     Given There is a release with channel "beta"
-    And The received version is "9.0.6"
+    And The received version is "9.0.7"
     When The request is sent
     Then The response is non-empty
-    And URL to download is "https://download.owncloud.org/community/owncloud-9.1.2.zip"
+    And URL to download is "https://download.owncloud.org/community/owncloud-9.1.3.zip"
 
   Scenario: Updating an outdated ownCloud 9.0.5 on the beta channel
```
Comment (davitol) [inline update-server/tests/integration/features/update.beta.feature]: There should be another similar scenario for 9.0.6 --> 9.0.7. @PVince81

Comment (PVince81) [inline update-server/tests/integration/features/update.beta.feature]: feel free to add it yourself

**Comment by VicDeo** [comment]
8.2.8->8.2.9->9.0.7 No issues

**Comment by VicDeo** [comment]
9.0.7 -> 9.1.3  Works too:
```
# sudo -uwww-data php occ integr:check-core
  - EXTRA_FILE:
    - container_status.php:
      - expected: 
      - current: bbba213c06b901c95b93af25400b14d9702d33ad8568724ed2e2f7e123d587de4cf7fc0656dc7201b031551c831792791b638f937c72833de1fb2874b74e21c7
    - feed.xml:
      - expected: 
      - current: bcce6f3192751fb645401554e088ec0a4a060693488497bb48eba4cbdb31912fa83ff557e8604057af372e92ffcac97edeeb19b89f8803192230b0e33272b729
```

**Comment by VicDeo** [comment]
@davitol 8.2.8->9.0.7 will not work.
8.2.9 has https://github.com/owncloud/updater/pull/384 fixing provisioning_api migration

**Comment by davitol** [comment]
> 9.0.7 -> 9.1.3 Works too:

@VicDeo Tested again and works, so if we are aware of

> 8.2.8->9.0.7 will not work.

then :+1:

**Comment by VicDeo** [comment]
ah. got it.
@PVince81 we might need to adjust `daily` channel logic. 
8.2.9 should be offered for all 8.2 that are lower than  `8.2.9` first and only if the current version is 8.2.9 or above it should migrate the most recent 9.1
(Just like other channels do)

**Comment by davitol** [comment]
@VicDeo @PVince81 What about this PR and this commit? https://github.com/owncloud/administration/pull/103/commits/d88a6a18875603304f17702ac32e9c252ac6d3ae

should we merge?

**Comment by PVince81** [comment]
Missed the mark again... now we can update this again to say 9.1.4.

@VicDeo please take over the updates

**Comment by PVince81** [comment]
@davitol @VicDeo let's try and make separate PRs for each series.

The trouble with the current approach is that if an issue is found with a path like 8.1->8.2, then it would also block 9.0->9.1 updates needlessly. If these were tested any worked then we should just push these to the server.

**Comment by PVince81** [comment]
Why does this PR have 1823 files changes ? That can't be right.

Closing.

@VicDeo please submit separate PRs invidiually for the different updates.

**Pull request closed without merging**